includes/models/class-reviews.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * @since x.x
+ */
+
+// Exit if accessed directly
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+class WPBDP_Reviews {
+
+	private $option_name = 'wpbdp_reviewed';
+
+	private $review_status = array();
+
+	private static $instance = null;
+
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Add admin notices as needed for reviews
+	 */
+	public function review_request() {
+
+		// Only show the review request to high-level users on business directory pages
+		if ( ! WPBDP_App_Helper::is_bd_page() ) {
+			return;
+		}
+		// Verify that we can do a check for reviews
+		$this->set_review_status();
+
+		// Check if it has been dismissed or if we can ask later
+		$dismissed = $this->review_status['dismissed'];
+		if ( $dismissed === 'later' && $this->review_status['asked'] < 3 ) {
+			$dismissed = false;
+		}
+
+		$week_ago = ( $this->review_status['time'] + WEEK_IN_SECONDS ) <= time();
+
+		if ( empty( $dismissed ) && $week_ago ) {
+			$this->review();
+		}
+	}
+
+	/**
+	 * When was the review request last dismissed?
+	 */
+	private function set_review_status() {
+		$user_id = get_current_user_id();
+		$review  = get_user_meta( $user_id, $this->option_name, true );
+		$default = array(
+			'time'      => time(),
+			'dismissed' => false,
+			'asked'     => 0,
+		);
+
+		if ( empty( $review ) ) {
+			// Set the review request to show in a week
+			update_user_meta( $user_id, $this->option_name, $default );
+		}
+
+		$review              = array_merge( $default, (array) $review );
+		$review['asked']     = (int) $review['asked'];
+		$this->review_status = $review;
+	}
+
+	/**
+	 * Maybe show review request
+	 */
+	private function review() {
+
+		// show the review request 3 times, depending on the number of entries
+		$show_intervals = array( 50, 200, 500 );
+		$asked          = $this->review_status['asked'];
+
+		if ( ! isset( $show_intervals[ $asked ] ) ) {
+			return;
+		}
+
+		$entries = WPBDP_Listing::count_listings();
+		$count   = $show_intervals[ $asked ];
+		$user    = wp_get_current_user();
+
+		// Only show review request if the site has collected enough entries
+		if ( $entries < $count ) {
+			// check the entry count again in a week
+			$this->review_status['time'] = time();
+			update_user_meta( $user->ID, $this->option_name, $this->review_status );
+
+			return;
+		}
+
+		if ( $entries <= 100 ) {
```

Comment:
I think we can move this `if ... else` to another method. We want to keep the function size smaller. And it would help a lot for readability and future changes.

(Steph recommended this book https://enos.itcollege.ee/~jpoial/oop/naited/Clean%20Code.pdf)